File: Scripts/laptop/call-an-object-method/java/CallAnObjectMethod.java
public class CallAnObjectMethod {

  private static volatile int r = 1; 
  public CallAnObjectMethod(){
  }

  public void someMethod(int value){
   this.r = value;
  }

	public static int executeTask(int i) {

		CallAnObjectMethod obj = new CallAnObjectMethod();
		obj.someMethod(1);
		return i + 1;
	}

public static void main(String[] args){

	for (int i = 0; i < 2000000000; ++i) {
		r = executeTask(i + r);
	}
}

}

File: Scripts/laptop/call-an-object-method/go/call-an-object-method.go
package main

type MyGoClass struct {
    variable  int // "instance variables"
}

func  someMethod(b *MyGoClass, i int) {
	obj_2 :=&MyGoClass{variable: b.variable + i}
	_ = obj_2
}

func executeTask(i int) int {

	obj := &MyGoClass{variable: 1000}
	someMethod(obj, i)
	return i;
}

func main(){
r := int(1)    

for i := 0; i < 2000000000; i++ {
	r = executeTask(i)
}
_=r

}
